olad/OlaServer.cpp

```diff
@@ -472,9 +476,15 @@ bool OlaServer::InternalNewConnection(
 }
 
 void OlaServer::ReloadPluginsInternal() {
+#ifdef HAVE_LIBSYSTEMD
+  ola::notify_systemd(0, "RELOADING=1\nSTATUS=Reloading plugins\n");
```

Comment:
Could the socket have gone away by the time this gets fired though? Potentially minutes or hours/days later?

My previous comment was more along the lines of if I'm trying to troubleshoot why my reload isn't getting to systemd, I'd want to see (at info/debug level) the positive return code from calling sd_notify, so I know it's the systemd end I need to look at, and not OLA, or vice versa.

Reply:
>Could the socket have gone away by the time this gets fired though? Potentially minutes or hours/days later?

Well, I'm not sure that this could _even_ happen, unless someone removes the environment variable. The socket is actually passed as a socket address (path name or name in abstract namespace, _a-la_ unix domain sockets), and `sd_notify` establishes a new socket sending data to that address, closing it on every return.

If `systemd` did somehow... fail, I think the whole system would be going up in flames, by then. But if it did happen and nobody is listening on the socket address, then the `sendmsg()` call in `sd_notify()` will fail and the error will be logged through `OLA_WARN`.

The only way for the `notify_systemd` call to fail silently is the removal of the `NOTIFY_SOCKET` environment variable.

Reply:
Okay, fair enough.